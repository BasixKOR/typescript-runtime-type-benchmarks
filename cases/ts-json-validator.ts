import { createSchema as S, TsjsonParser } from 'ts-json-validator'
import { Data } from '../data'

const parser = new TsjsonParser(
  S({
    type: 'object',
    properties: {
      number: S({ type: 'number' }),
      neg_number: S({ type: 'number', maximum: 0 }),
      max_number: S({ type: 'number' }),
      string: S({ type: 'string' }),
      long_string: S({ type: 'string', minLength: 100 }),
      boolean: S({ type: 'boolean' }),
      deeplyNested: S({
        type: 'object',
        properties: {
          foo: S({ type: 'string' }),
          num: S({ type: 'number' }),
          bool: S({ type: 'boolean' })
        }
      })
    },
  })
)

export function caseTsJsonValidator(data: Data) {
  if (parser.validates(data)) {
    return data
  }
  throw new Error('Invalid')
}
