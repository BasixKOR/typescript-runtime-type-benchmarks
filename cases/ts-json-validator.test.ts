import { caseTsJsonValidator } from './ts-json-validator'
import { DATA } from '../data'

describe('ts-json-validator', () => {
  it('should pass', () => {
    const res = caseTsJsonValidator(DATA)
    expect(res).toEqual(DATA)

    if (res) {
      expect(res.number).toEqual(DATA.number)
    }
  })

  it('should fail negative number test', () => {
    const req = Object.assign({}, DATA, { neg_number: 1 })
    expect(() => caseTsJsonValidator(req)).toThrowErrorMatchingSnapshot()
  })

  it('should fail min length test', () => {
    const req = Object.assign({}, DATA, { long_string: '' })
    expect(() => caseTsJsonValidator(req)).toThrowErrorMatchingSnapshot()
  })
})
