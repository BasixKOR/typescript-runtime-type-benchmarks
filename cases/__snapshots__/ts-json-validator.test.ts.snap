// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-json-validator should fail min length test 1`] = `"Invalid"`;

exports[`ts-json-validator should fail negative number test 1`] = `"Invalid"`;
